virtinst/install/urldetect.py

```diff
@@ -851,6 +854,41 @@ def _set_manual_kernel_paths(self):
                  self.cache.libosinfo_mediaobj.get_initrd_path())
         ]
 
+class _OracleLinuxDistro(_DistroTree):
+    """
+    Special distro that has kernel, initrd under images/pxeboot
+    """
+    PRETTY_NAME = "Oracle Linux"
+    matching_distros = ["ol"]
+    _variant_prefix = "ol"
+
+    @classmethod
+    def is_valid(cls, cache):
+        if cache.treeinfo_family_regex(".*Oracle Linux.*"):
+            return True
+
+    def _set_manual_kernel_paths(self):
+        self._kernel_paths += [
+            ("images/pxeboot/vmlinuz",
+             "images/pxeboot/initrd.img")]
```

Comment:
note that this is needed only if the `.treeinfo` is not matched, and the `is_valid()` method seems to assume it will -- see `_DistroTree.__init__()`